src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -1654,11 +1909,40 @@ public static void TryAddLifecycleHook<T>(this Microsoft.Extensions.DependencyIn
 
 namespace Aspire.Hosting.Publishing
 {
+    public sealed partial class AfterPublishEvent : Eventing.IDistributedApplicationEvent
```

Comment:
@mitchdenny - Should these be `[Experimental]`?

Reply:
No ... these events are safe to not be experimental. We'll be firing them off regardless of how the publishing system itself evolves because there will always be a moment in apphost control where we start and stop publishing.